### DISCUSSION TITLE
[Improvement] Consider changing parameters positions on rebind function

### DISCUSSION THREAD

**Discussion by crisadamo** [modular/modular#180] [General]
It will be nice to change the current `rebind` parameters from `[dest, src]` to `[src, dest]` since its more intuitive that the other way around.

The current signature is `rebind[dest_type, src_type](src_val)`

**Comment by lattner** [comment]
The current way works better with parameter inference though, because you can call it with `rebind[dest_type](src_val)` and have src_type inferred from the argument.

**Reply by czheo** [reply]
Wondering if it is a coincidence that assembly like `mov dest, src` also has the same order...
